pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -0,0 +1,610 @@
+		newConfigs = *lb.Properties.FrontendIPConfigurations
+	}
+	if !wantLb {
+		for i := len(newConfigs) - 1; i >= 0; i-- {
+			config := newConfigs[i]
+			if strings.EqualFold(*config.Name, lbFrontendIPConfigName) {
+				glog.V(3).Infof("reconcile(%s)(%t): lb frontendconfig(%s) - dropping", serviceName, wantLb, lbFrontendIPConfigName)
+				newConfigs = append(newConfigs[:i], newConfigs[i+1:]...)
+				dirtyConfigs = true
+			}
+		}
+	} else {
+		foundConfig := false
+		for _, config := range newConfigs {
+			if strings.EqualFold(*config.Name, lbFrontendIPConfigName) {
+				foundConfig = true
+				break
+			}
+		}
+		if !foundConfig {
+			newConfigs = append(newConfigs,
+				network.FrontendIPConfiguration{
+					Name: to.StringPtr(lbFrontendIPConfigName),
+					Properties: &network.FrontendIPConfigurationPropertiesFormat{
+						PublicIPAddress: &network.PublicIPAddress{
+							ID: pip.ID,
+						},
+					},
+				})
+			glog.V(10).Infof("reconcile(%s)(%t): lb frontendconfig(%s) - adding", serviceName, wantLb, lbFrontendIPConfigName)
+			dirtyConfigs = true
+		}
+	}
+	if dirtyConfigs {
+		dirtyLb = true
+		lb.Properties.FrontendIPConfigurations = &newConfigs
+	}
+
+	// update probes/rules
+	expectedProbes := make([]network.Probe, len(service.Spec.Ports))
+	expectedRules := make([]network.LoadBalancingRule, len(service.Spec.Ports))
+	for i, port := range service.Spec.Ports {
+		lbRuleName := getRuleName(service, port)
+
+		transportProto, _, probeProto, err := getProtocolsFromKubernetesProtocol(port.Protocol)
+		if err != nil {
+			return lb, false, err
+		}
+
+		expectedProbes[i] = network.Probe{
+			Name: &lbRuleName,
+			Properties: &network.ProbePropertiesFormat{
+				Protocol:          probeProto,
+				Port:              to.Int32Ptr(port.NodePort),
+				IntervalInSeconds: to.Int32Ptr(5),
+				NumberOfProbes:    to.Int32Ptr(2),
+			},
+		}
+
+		expectedRules[i] = network.LoadBalancingRule{
+			Name: &lbRuleName,
+			Properties: &network.LoadBalancingRulePropertiesFormat{
+				Protocol: transportProto,
+				FrontendIPConfiguration: &network.SubResource{
+					ID: to.StringPtr(lbFrontendIPConfigID),
+				},
+				BackendAddressPool: &network.SubResource{
+					ID: to.StringPtr(lbBackendPoolID),
+				},
+				Probe: &network.SubResource{
+					ID: to.StringPtr(az.getLoadBalancerProbeID(lbName, lbRuleName)),
+				},
+				FrontendPort: to.Int32Ptr(port.Port),
+				BackendPort:  to.Int32Ptr(port.NodePort),
+			},
+		}
+	}
+
+	// remove unwated probes
+	dirtyProbes := false
+	var updatedProbes []network.Probe
+	if lb.Properties.Probes != nil {
+		updatedProbes = *lb.Properties.Probes
+	}
+	for i := len(updatedProbes) - 1; i >= 0; i-- {
+		existingProbe := updatedProbes[i]
+		if serviceOwnsRule(service, *existingProbe.Name) {
+			glog.V(10).Infof("reconcile(%s)(%t): lb probe(%s) - considering evicting", serviceName, wantLb, *existingProbe.Name)
+			keepProbe := false
+			for _, expectedProbe := range expectedProbes {
```

Comment:
looks to me like you can extract a `findProbe` method and use it here and below (if not other places)